service/src/main/kotlin/fi/espoo/evaka/reports/FuturePreschoolersReport.kt

```diff
@@ -0,0 +1,196 @@
+                    )
+                    it.setStatementTimeout(REPORT_STATEMENT_TIMEOUT)
+                    it.getFuturePreschoolerRows()
+                }
+            }
+            .also { Audit.FuturePreschoolers.log(meta = mapOf("count" to it.size)) }
+    }
+
+    @GetMapping("/reports/future-preschoolers/groups")
+    fun getPreschoolGroups(
+        db: Database,
+        user: AuthenticatedUser,
+        clock: EvakaClock,
+        @RequestParam municipal: Boolean
+    ): List<PreschoolGroupsReportRow> {
+        return db.connect { dbc ->
+                dbc.read {
+                    accessControl.requirePermissionFor(
+                        it,
+                        user,
+                        clock,
+                        Action.Global.READ_FUTURE_PRESCHOOLERS
+                    )
+                    it.setStatementTimeout(REPORT_STATEMENT_TIMEOUT)
+                    it.getPreschoolGroupsRows(municipal)
+                }
+            }
+            .also { Audit.FuturePreschoolers.log(meta = mapOf("count" to it.size)) }
+    }
+}
+
+const val preschoolSelectionAge = 5
+
+fun Database.Read.getFuturePreschoolerRows(): List<FuturePreschoolersReportRow> =
+    createQuery<DatabaseTable> {
+            sql(
+                """
+SELECT p.id, 
+    p.social_security_number AS childSsn,
+    p.last_name AS childLastName,
+    p.first_name AS childFirstName,
+    p.street_address AS childAddress,
+    p.postal_code AS childPostalCode,
+    upper(p.post_office) AS childPostOffice,
+    d.name AS unitName,
+    d.street_address AS unitAddress,
+    d.postal_code AS unitPostalCode,
+    upper(d.post_office) AS unitPostOffice,
+    NULL as unitArea, -- cannot be obtained from current evaka dataset, care area is not precise enough
+    g1.last_name AS guardian1LastName,
+    g1.first_name AS guardian1FirstName,
+    g1.street_address AS guardian1Address,
+    g1.postal_code AS guardian1PostalCode,
+    upper(g1.post_office) AS guardian1PostOffice, 
+    g1.phone AS guardian1Phone,
+    g1.email AS guardian1Email,
+    g2.last_name AS guardian2LastName, 
+    g2.first_name AS guardian2FirstName, 
+    g2.street_address AS guardian2Address,
+    g2.postal_code AS guardian2PostalCode,
+    upper(g2.post_office) AS guardian2PostOffice, 
+    g2.phone AS guardian2Phone,
+    g2.email AS guardian2Email,
+    CASE WHEN sn.shift_care = 'FULL'::shift_care_type OR sn.shift_care = 'INTERMITTENT'::shift_care_type THEN true ELSE false END AS shiftCare,
+    NULL AS languageEmphasisGroup, -- cannot be obtained from current evaka dataset
+    CASE WHEN sno.name_fi like 'Kaksivuotinen%' THEN true ELSE false END AS twoYearPreschool
+FROM person p
+JOIN placement pl ON pl.child_id = p.id
+JOIN daycare d ON d.id = pl.unit_id
+LEFT JOIN (
+    SELECT id, last_name, first_name, street_address, postal_code, post_office, phone, email
+    FROM person
+) g1 ON g1.id = (SELECT head_of_child FROM fridge_child WHERE child_id = p.id)
```

Comment:
Head of child (päämies) is not the same as guardian. If you want guardian (huoltaja), use the `guardian` table. Also consider whether you may want foster parents (sijaisvanhempi) in the `foster_parent` table.

If head of child is what you want, note that it's timeseries data, so you should take the one where the date of interest falls in the `start_date` - `end_date` range.